src/Lumina/Excel/ExcelSheet.cs

```diff
@@ -36,14 +37,22 @@ public class ExcelSheet< T > : ExcelSheetImpl, IEnumerable< T > where T : ExcelR
                 return cachedRow;
             }
 
-            var parser = GetRowParser( row, subRow );
+            var page = GetPageForRow( row );
+            if( page == null )
+            {
+                return null;
+            }
+            
+            var parser = GetRowParser( page, row, subRow );
             if( parser == null )
             {
                 return null;
             }
             
             var rowObj = Activator.CreateInstance< T >();
-            rowObj.PopulateData( parser, GameData, RequestedLanguage );
+            
+            lock(page.File.ReaderLock)
```

Comment:
i was thinking a while back it would probably better to just construct a new reader for each rowparser but i guess this would work

can you put curly braces in and fix the spacing tho?

Reply:
Fixed formatting, that'd be a better solution for the long run but we can do that once someone gets into a refactoring mood